packages/storage-plugin/src/storage.module.ts

```diff
@@ -20,7 +22,14 @@ export function storageOptionsFactory(options: NgxsStoragePluginOptions) {
   };
 }
 
-export function engineFactory(options: NgxsStoragePluginOptions): StorageEngine | null {
+export function engineFactory(
+  options: NgxsStoragePluginOptions,
+  platformId: any
+): StorageEngine | null {
+  if (isPlatformServer(platformId)) {
```

Comment:
Did you check if it's working?

Reply:
Yes, it works. I'll check in my test project so you can easily try it.

Reply:
Hm, I just copied your code, built package, copied bundle and it prints `undefined`:
```TS
console.log(`platformId = ${platformId}`);
if (isPlatformServer(platformId)) {
  return new InMemoryStorageEngine();
}
```

Reply:
Also the problem is that you can easily just return `null` as `InMemoryStorageEngine` seems meaningless for me. What's the purpose of it? You can also inject `PLATFORM_ID` into plugin and check if it's executed on the server and just return `next(...)`:
```TS
handle(state: any, event: any, next: NgxsNextPluginFn) {
  if (isPlatformServer(this.platformId)) {
    return next(state, event);
  }

  ...
}
```

Reply:
if `InMemoryStorageEngine` is created on the server side which is bootstrapped only once - this means that the instance will be the same for all requests right?

Reply:
![image](https://user-images.githubusercontent.com/255721/59066276-f190da80-88ae-11e9-9c23-c89a6d304f34.png)
So this should work.

If you return `null` instead of the `InMemoryStorageEngine` the build will fail because no storage engine is selected (so you would have to manually provide one via `app.server.module.ts`

Injecting it into plugin and nexting is a different behaviour, in the long run I want to use the storage plugin server side and provice a custom storage engine to pre-initialize the store for server side rendering with defaults (e.g. theme, language etc.) so just skipping the plugin would not be ideal.

About being created only once, in my setup every request re-creates the storage engine (checking with debugger), but maybe there is something about SSR I am not aware about.